credential/generate.go

```diff
@@ -0,0 +1,85 @@
+package credential
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/jsonld"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/suite/ecdsasecp256k1signature2019"
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+)
+
+// Generator is a verifiable credential generator.
+type Generator struct {
+	vc        Descriptor
+	signer    verifiable.Signer
+	signerDID string
+	parser    *DefaultParser
+}
+
+// New allow to Generate a verifiable credential with the given credential descriptor.
+// Example:
+//
+//	vc, err := credential.New(
+//	    template.NewGovernance().
+//	        WithDatasetDID("did:key:...").
+//	        WithGovAddr("axone1234..."),
+//	).
+//	WithParser(parser).
+//	WithSignature(signer, "did:key:..."). // Signature is optional and Generate a not signed VC if not provided.
+//	Generate()
+func New(descriptor Descriptor) *Generator {
```

Comment:
Maybe we could follow the functional options pattern this by providing options to configure the generator at its creation, it'd be a more idiomatic go code, for example:

```golang
func New(descriptor Descriptor, opts ...Option) *Generator {
	g := &Generator{
		vc: descriptor,
	}
	for _, opt := range opts {
		opt(g)
	}
	return g
}

// Option is a function that configures a Generator.
type Option func(*Generator)

func WithParser(parser *DefaultParser) Option {
	return func(g *Generator) {
		g.parser = parser
	}
}

func WithSigner(signer verifiable.Signer, did string) Option {
	return func(g *Generator) {
		g.signer = signer
		g.signerDID = did
	}
}
```

Reply:
Nice, I will do that for all descriptors too !